Sources/TerminalAPIKit/Helpers/DataHelpers.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  DataHelpers.swift
+//  TerminalAPIKit
+//
+//  Copyright (c) 2022 Adyen N.V.
+//
+
+import Foundation
+
+internal extension Data {
+    
+    /// Creates a new data buffer with securely generated random bytes.
+    ///
+    /// - Parameter count: The number of randomly generated bytes.
+    init?(secureRandomBytesWithCount count: Int) {
+        self.init(count: count)
+        
+        let result = withUnsafeMutableBytes { bytes -> OSStatus in
+            guard let baseAddress = bytes.baseAddress else {
+                return errSecAllocate
+            }
+            
+            return SecRandomCopyBytes(kSecRandomDefault, count, baseAddress)
+        }
+        
+        if result != errSecSuccess {
+            return nil
+        }
+    }
```

Comment:
```suggestion
    init?(secureRandomBytesWithCount count: Int) {
        self.init(count: count)
        
        var bytes = [UInt8]()
        bytes.reserveCapacity(count)
        
        let result = SecRandomCopyBytes(kSecRandomDefault, count, &bytes)
        
        if result != errSecSuccess {
            return nil
        }
    }
```

Reply:
fixed this one as well

Reply:
i don't think this is correct. we never actually assign the `bytes` generated with `SecRandomCopyBytes`.

Reply:
fixed